core/src/main/scala/chisel3/internal/Binding.scala

```diff
@@ -123,3 +125,5 @@ sealed trait LitBinding extends UnconstrainedBinding with ReadOnlyBinding
 case class ElementLitBinding(litArg: LitArg) extends LitBinding
 // Literal binding attached to the root of a Bundle, containing literal values of its children.
 case class BundleLitBinding(litMap: Map[Data, LitArg]) extends LitBinding
+// Literal binding attached to the root of a Vec, containing literal values of its children.
+case class VecLitBinding(litMap: mutable.LinkedHashMap[Data, LitArg]) extends LitBinding
```

Comment:
I don't think this should be mutable. It might make sense to use mutable to build the final data structure, but the stored data structure should be immutable.

Reply:
Switched after generation to immutable.ListMap, important to maintain order